pkg/cloud/services/ec2/securitygroups.go

```diff
@@ -455,7 +456,7 @@ func (s *Service) getSecurityGroupIngressRules(role infrav1.SecurityGroupRole) (
 }
 
 func (s *Service) getSecurityGroupName(clusterName string, role infrav1.SecurityGroupRole) string {
-	return fmt.Sprintf("%s-%v", clusterName, role)
+	return fmt.Sprintf("%s-%v", strings.TrimPrefix(clusterName, "sg-"), role)
```

Comment:
```suggestion
    groupPrefix := clusterName
	if strings.HasPrefix(clusterName, "sg-") {
		groupPrefix = "@" + clusterName 
	}
	return fmt.Sprintf("%s-%v", groupPrefix, role)
```

Reason for this is that Kubernetes names are DNS RFC-1123, and AWS allows 
> Allowed characters are a-z, A-Z, 0-9, spaces, and ._-:/()#,@[]+=;{}!$*.
Prefixing with @ for the `sg-` special case won't conflict with the same name without the `sg-` prefix.